packages/dom/src/lib/ElementAssertion.ts

```diff
@@ -90,4 +90,48 @@ export class ElementAssertion<T extends Element> extends Assertion<T> {
       invertedError,
     });
   }
+
+/**
+ * Check if the element has a specific class or classes.
+ *
+ * Validates that the provided element contains specified classes.
+ * Allows checking for one or more class names and supports exact matching.
+ *
+ * @param classNames - A single class name or an array of class names to check.
+ * @param options - Optional settings for matching:
+ *  - `exact` (boolean): When true, checks for an exact match of all classes.
+ * @returns the assertion instance.
+ */
+  public toHaveClass(classNames: string | string[], options: { exact?: boolean; } = {}): this {
```

Comment:
What do you think about making more than one assertion instead of adding the options param? To me, it'll provide more readability if we have the following:

```ts
expect(elem).toHaveClass("w-full"); // single arg

expect(elem).toHaveAnyClass("w-full", "flex"); // variadic args

expect(elem).toHaveAllClasses("w-full", "flex", "gap-8"); // variadic args
```